docs/content/topics/operators/gateways.md

```diff
@@ -11,340 +11,421 @@ aliases:
 
 # Brigade Gateways
 
-This guide explains how gateways work, and provides guidance for creating your own
-gateway.
+This guide explains how gateways work, and provides guidance for creating your
+own.
 
 ## What Is A Brigade Gateway?
 
-The [Brigade architecture](design.md) is oriented around the concept that Brigade
-scripts run as a response to one or more events. In Brigade, a _gateway_ is an
-entity that generates events. Often times, it translates some external trigger
-into a Brigade event.
+The [Brigade architecture](/topics/design) is oriented around the concept that
+Brigade scripts run as a response to one or more events. In Brigade, a gateway
+is an entity that generates events. Often times, it translates some external
+trigger into a Brigade event.
 
-Brigade ships with the ability to enable various gateways that are ready to go.
+While there is no default gateway that ships with Brigade, installing one
+alongside Brigade is as simple as a [Helm] chart install, along with any
+additional setup particular to a given gateway. See
+[below](#available-gateways) for a current listing of v2-compatible gateways.
 
-These include the [Container Registry Gateway](dockerhub.md), the [Github Gateway](./github.md)
-and the [Generic Gateway](./genericgateway.md).  They can all be enabled via top-level
-Helm [chart flags](https://github.com/brigadecore/charts/blob/master/charts/brigade/values.yaml).
-
-All of these provide HTTP-based listeners that receive incoming requests
-(from a container registry, Github or other platforms and systems) and generate
-Brigade events as a result.
+All of these provide HTTP(S)-based listeners that receive incoming requests
+(from Github or other platforms and systems) and generate Brigade events as a
+result.
 
 However, Brigade's gateway system works with more than just webhooks.
 
-For example, the `brig` client also acts as a gateway. When you execute a `brig run`
-command, `brig` creates a Brigade event. By default, it emits an `exec` event. And
-Brigade itself processes this event no differently than it processes the GitHub
-or container registry events.
+For example, the `brig` client behaves similarly to a gateway. When you execute
+a `brig event create` command, `brig` creates a Brigade event of source
+`brigade.sh/cli` and of type `exec`. Brigade itself processes this event no
+differently than it processes events from gateways tracking external activity.
 
 There are no rules about what can be used as a trigger for an event. One could
-write a gateway that listens on a message queue, or runs as a chat bot, or watches
-files on a filesystem... any of these could be used to trigger a new Brigade event.
+write a gateway that listens on a message queue, or runs as a chat bot, or
+watches files on a filesystem... any of these could be used to trigger a new
+Brigade event.
```

Comment:
I really like this section. I'm wondering if it should maybe move up higher (because I think it's super important) and potentially replace the bit about our (current) gateways being HTTP(S) based...

I think the fact that gateways could hypothetically receive triggers through a variety of means trumps the fact that the ones we happen to have implemented so far use HTTP(S)...

Or maybe we could word something like this?

> Currently, all of our officially supported gateways are triggered by webhooks delivered from external systems over HTTP(S), but there are no rules about what can be used as a trigger for an event. A gateway could hypothetically listen on a message queue, run as a chat bot, or watch files on a file system.

Reply:
This wording could also solve the problem of the term "webhooks" not being well-introduced, as mentioned in another comment.

Reply:
Updated and re-org'd... PTAL when convenient.